delegated-staking.md

```diff
@@ -0,0 +1,171 @@
+  - **Parameters**: 
+     - `delegatee`:  the delegatee a user staked to. 
+ - **When**: Any time
+ - **Behavior**: Aggregates all `delegation`s of the caller for the particular `delegatee`, accumulates all the amounts and rewards, transfers them to the caller and removes these `delegation`s from the list of `delegations` to that particular `delegatee`.
+
+6. **`update_delegatees_stake()`** 
+ - **Description**: Sets the correct amount a `delegatee` has staked in the main staking logic.
+  - **Parameters**: None.
+ - **When**: Every time a staker in the main staking logic, who also registered as a `delegatee`, changes his staked amount in any way (stake, adjust, unstake)
+ - **Behavior**: In the list of all `delegations` for this `delegatee`, finds the `delegation` belonging to the `delegatee`, and adjusts the field `staked_amount` accordingly.
+
+  ---
+
+### **Pending investigations**
+
+1. In the above-described logic, delegated funds count directly to the stakers *election weight* (proper term required). Staker's funds are required to be locked for a certain amout of epochs. This is diametral to the current delegation logic of allowing `delegated stakes` to be contributed and withdrawn at any time. 
+
+    For the contributing: Where is what kind of additional logic needed to somehow delay the attribution of the delegated funds to the staking power of a block producer, or, somehow otherwise take the necessity of a constant amount of funds to be locked for some epochs into account?
+
+    For the withdrawing: What checks and conditions could be applied to the `withdraw_delegated_stakes` function to prevent delegations to be withdrawn at an epoch they are supposed to be still locked, semantically as part of a locked block producer's stake?
+    
+2. Although `split_reward_to_delegators()` only distributes funds nominally instead of transfering actual value, it is to be verified whether the call to it has actually `value` assigned to it or not, therefore adding / removing the `payable` modifier might be necessary.
+  
+  
+### **Contextual constraints**
+ The above delegation logic assumes the following requirements to the `main staking logic` to be met: 
+ 
+ 1. On any adjustment of a staker's amount, who registered as a `delegatee`, `update_delegatees_stake()` is called to keep the delegation logic's book-keeping in sync with the amount of funds staked by stakers in the `main staking logic`. By checking if the Caller is a `delegatee`, all delegatee's calls to adjust their stake shall be forwarded *to the delegation logic* (as for example the direct withdrawal from the staker's total stake in the `main staking logic` would withdraw also the funds *delegated* to him). The delegation logic has means of adjusting the staked amount in the `main staking logic` in accordance to the changes in its book keeping, according stubs are included in the current code.
```

Comment:
This is not settled, but we are leaning towards the main staking contract not knowing if the logic behind a StakingValidator is a single entity or a delegatee backed by several delegates.